### PR TITLE
Health 

### PR DESCRIPTION
**Pull request by lukanus**
Very initial metric that gives you ability to add basic metrics (this one with database size) automagicaly to every project we have.

### CODE REVIEW

File: health/database/database.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+		Desc:      "Duration how long it takes to ping",
+		Tags:      []string{"type"},
+	})
+
+	SizeMetric = metrics.MustNewGaugeWithTags(metrics.Options{
+		Namespace: "indexerhealth",
+		Subsystem: "database",
+		Name:      "size",
+		Desc:      "Current size of database",
+		Tags:      []string{"type"},
```
Comment (pacmessica) [inline health/database/database.go:19]: "type" sounds very generic to me. how is this used?

File: health/database/postgreshealth/postgreshealth.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+type PingCheck struct {
+	On       time.Time     `json:"on"`
+	Duration time.Duration `json:"duration"`
+	Status   string        `json:"status"`
+	Error    string        `json:"error"`
+}
+
+type SizeCheck struct {
+	On     time.Time `json:"on"`
+	Size   uint64    `json:"duration"`
```
Comment (pacmessica) [inline health/database/postgreshealth/postgreshealth.go:30]: check the json label :p

File: health/database/database.go (hunk 1 above)
Comment (lukanus) [reply health/database/database.go:19]: type of the database, "postgres" in this particular case, I'm assuming we may have more db's in future

File: health/database/postgreshealth/postgreshealth.go (hunk 2 above)
Comment (lukanus) [reply health/database/postgreshealth/postgreshealth.go:30]: Yeah I just realized I haven't pushed my last commit with this ;D

File: health/database/database.go (hunk 1 above)
Comment (pacmessica) [reply health/database/database.go:19]: should the tag be "postgres" then?

Comment (lukanus) [reply health/database/database.go:19]: https://github.com/figment-networks/indexing-engine/pull/9/files#diff-5b5437160ac385fd0a6e1e3e25c7d0670f78201dd13d324b42157eaba5bcf2fdR48

File: health/database/database.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package database
+
+import "github.com/figment-networks/indexing-engine/metrics"
+
+var (
+	PingMetric = metrics.MustNewHistogramWithTags(metrics.HistogramOptions{
+		Namespace: "indexerhealth",
```
Comment (pienkowb) [inline health/database/database.go:7]: How about using `indexer` as the namespace everywhere? I used that for the pipeline metrics and we could also use it here. That would give us well-sounding metric names like `indexer_pipeline_task_duration`, `indexer_database_size` etc.

File: health/database/database.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+		Subsystem: "database",
+		Name:      "ping",
+		Desc:      "Duration how long it takes to ping",
+		Tags:      []string{"type"},
+	})
+
+	SizeMetric = metrics.MustNewGaugeWithTags(metrics.Options{
+		Namespace: "indexerhealth",
```
Comment (pienkowb) [inline health/database/database.go:15]: Same here.

File: health/database/database.go (hunk 1 above)
Comment (pienkowb) [reply health/database/database.go:19]: Maybe we could call this `system`? But again, it's quite generic.

Comment (pienkowb) [reply health/database/database.go:19]: BTW, I don't think that knowing the database type is crucial for monitoring the size of an indexer's database. Maybe we don't even need this tag?

**Review by pienkowb — COMMENTED** [review]
Great work, Łukasz 👍 I've added some of my thoughts.

**Comment by pacmessica** [comment]
@lukanus hey! so the pipeline metrics PR is merged. It'd be nice if we could finish this one up to, then when fixing up metrics in existing pipelines we can implement this as well

File: health/database/database.go (hunk 3 above)
Comment (lukanus) [reply health/database/database.go:7]: We can use this in other, more generic ones. Do we have the docs about it somewhere? I would vote for making it "figment_health" or something like that, so you can attach it everywhere. @pacmessica @pienkowb wdyt?

File: health/database/database.go (hunk 1 above)
Comment (lukanus) [reply health/database/database.go:19]: The way of how that's composed is that those metrics are declared on different package than postgres, so it has to  attach db name somehow :) that's the reason why it's declared like that

File: health/database/database.go (hunk 3 above)
Comment (pienkowb) [reply health/database/database.go:7]: In that case, I'd go for `health`. Prefixing everything with `figment_` is a separate task and it should be done for all of our metrics (if we decide to do that).

Comment (pienkowb) [reply health/database/database.go:7]: BTW, if these are generic metrics applicable anywhere, maybe that should be a package on its own.

File: health/database/database.go (hunk 1 above)
Comment (pienkowb) [reply health/database/database.go:19]: If it's necessary to have this tag, I'd vote for `database_type` or `database_system`.

File: health/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- `/health` - Endpoint that always return 200 for dumb health check
+- `/readiness` - Endpoint meant to return information from different probes and return meaningful information about is service ready for connecting it to the traffic.
+
+## Metrics
+Service produces a set of metrics
+
+### Database
+- `health_database_ping` - Duration how long it takes to ping the database (float seconds)
+    - Tags:
+        - `database_type` - The name od database type (eg. `postgres`)
```
Comment (pienkowb) [inline health/README.md:18]: od => of

File: health/README.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+- `health_database_size` - Current size of database
+    - Tags:
+        - `database_type` - The name od database type (eg. `postgres`)
```
Comment (pienkowb) [inline health/README.md:21]: od => of

**Review by pienkowb — COMMENTED** [review]
@lukanus Let's fix the typos and we're good to go 👍

**Review by pienkowb — APPROVED** [review]

**Pull request merged**